### PR TITLE
Shader example

### PR DESCRIPTION
**Pull request by QuantumEntangledAndy**
This adds a shader example. This example focuses on:

- loading a uniform
- binding a uniform
- compiling a shader
- running a shader

The shader just colors a cube with a gradient between two colors

Screenshot 2022-01-01 at 21 48 39

### CODE REVIEW

File: examples/shader/shader.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        let gradient_buffer = globals
+            .get::<GradientBuffer>()
+            .expect("Gradient buffer must be loaded");
+
+        // Get the uniform that holds the camera's project view
+        let proj_view = globals
+            .get::<ProjView>()
+            .expect("ProjView buffer must be loaded");
+
+        // Set the shader on the pipline
```
Comment (Lowentwickler) [inline examples/shader/shader.rs:103]: typo: pipeline

File: examples/shader/shader.wgsl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return out;
+}
+
+// STAGE: FRAGMENT -------------------------------------------------------------------------------
+struct Gradient {
+  zenith_color: vec4<f32>;
+  nadir_color: vec4<f32>;
+};
+[[group(0), binding(1)]]
+var<uniform> u_skybox: Gradient;
```
Comment (Lowentwickler) [inline examples/shader/shader.wgsl:29]: Most likely the variable name must be `u_gradient`

**Review by Lowentwickler — CHANGES_REQUESTED** [review]
Awesome! This one was really wanted. There are just few small requests from my side.

Could you please also update the `Examples` section in `README.md` as a part of this PR?

**Comment by Lowentwickler** [comment]
@tree786 I bet you would love to see it

**Comment by Lowentwickler** [comment]
@QuantumEntangledAndy can we chat somewhere? My discord  ID is  Taurasch#4633 if you use it.

**Comment by QuantumEntangledAndy** [comment]
Sure just sent you a request on discord it is Andy#6008

File: examples/shader/shader.rs (hunk 1 above)
Comment (QuantumEntangledAndy) [reply examples/shader/shader.rs]: Done

File: examples/shader/shader.wgsl (hunk 2 above)
Comment (QuantumEntangledAndy) [reply examples/shader/shader.wgsl]: Done

**Comment by QuantumEntangledAndy** [comment]
Anything else on this one?

**Review by Lowentwickler — CHANGES_REQUESTED** [review]
One more thing and lets merge it.

File: examples/shader/main.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    camera.distance = 2.0;
+    camera.xz_angle = 0.0;
+
+    let mut mesh = cube::cube(0.5);
+
+    mesh.load(&renderer);
+
+    let mesh_handle = assets.store(mesh);
+
+    // Spawn skybox
```
[the remark this answers is not in the document]
Comment (QuantumEntangledAndy) [reply examples/shader/main.rs]: Done

**Pull request merged**